persistence-api/src/test/java/io/stargate/db/cdc/SchemaAwareCDCProducerTest.java

```diff
@@ -0,0 +1,179 @@
+/*
+ * Copyright The Stargate Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.stargate.db.cdc;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+import static org.junit.jupiter.params.provider.Arguments.arguments;
+
+import edu.umd.cs.findbugs.annotations.NonNull;
+import io.stargate.db.schema.Column;
+import io.stargate.db.schema.ImmutableTable;
+import io.stargate.db.schema.Table;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.stream.Stream;
+import org.apache.cassandra.stargate.db.CellValue;
+import org.apache.cassandra.stargate.db.MutationEvent;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.MethodSource;
+
+class SchemaAwareCDCProducerTest {
+  @Test
+  public void shouldUpdateSchemaWhenPublishNewTable() {
+    // given
+    MockSchemaAwareCDCProducer producer = new MockSchemaAwareCDCProducer();
+    Table table = createTable();
+
+    // when
+    producer.publish(createMutationEvent(table));
+
+    // then
+    assertThat(producer.createNumberOfCalls).isEqualTo(1);
+    assertThat(producer.sendNumberOfCalls).isEqualTo(1);
+  }
+
+  @Test
+  public void shouldCallCreateTableSchemaOnceForSameTableCalls() {
```

Comment:
nit: It looks like this test completely covers `shouldUpdateSchemaWhenPublishNewTable`... do we need both of them?

Reply:
It does but I think those are different functionalities. For the potential reader, it would be beneficial to have an additional test covering only this first happy-path